src/System.Private.CoreLib/shared/System/IO/StreamHelpers.CopyValidation.cs

```diff
@@ -42,5 +46,44 @@ public static void ValidateCopyToArgs(Stream source, Stream destination, int buf
                 throw new NotSupportedException(SR.NotSupported_UnwritableStream);
             }
         }
+
+        public static void ValidateCopyToArgs(Stream source, ReadOnlySpanAction<byte, object?> callback, int bufferSize)
+        {
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            ValidateCopyToArgsCore(source, bufferSize);
+        }
+
+        public static void ValidateCopyToArgs(Stream source, Func<ReadOnlyMemory<byte>, object?, CancellationToken, ValueTask> callback, int bufferSize)
+        {
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            ValidateCopyToArgsCore(source, bufferSize);
+        }
+
+        private static void ValidateCopyToArgsCore(Stream source, int bufferSize)
+        {
+            if (bufferSize <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bufferSize), bufferSize, SR.ArgumentOutOfRange_NeedPosNum);
+            }
+
+            bool sourceCanRead = source.CanRead;
+            if (!sourceCanRead && !source.CanWrite)
+            {
+                throw new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
+            }
+
+            if (!sourceCanRead)
+            {
+                throw new NotSupportedException(SR.NotSupported_UnreadableStream);
+            }
```

Comment:
The above 10 lines can be simplified to just:
```C#
if (!source.CanRead)
{
    throw source.CanWrite ? (Exception)
        new NotSupportedException(SR.NotSupported_UnreadableStream) :
        new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
}
```

Reply:
Fixed.